cdk/lib/frontend.ts

```diff
@@ -18,5 +34,360 @@ export class Frontend extends GuStack {
+        },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      userData,
+      roleConfiguration: {
+        additionalPolicies: policies,
+      },
+      scaling: {
+        PROD: {
+          minimumInstances: 3,
+          maximumInstances: 6,
+        },
+        CODE: {
+          minimumInstances: 1,
+          maximumInstances: 2,
+        },
+      },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
+    });
+
+    // TODO: remove this tag after the migration
+    const ec2AppAsg = ec2App.autoScalingGroup;
+    Tags.of(ec2AppAsg).add("gu:riffraff:new-asg", "true");
+
+    // ---- Alarms ---- //
+
+    const alarmName = (shortDescription: string) =>
+      `URGENT 9-5 - ${this.stage} ${shortDescription}`;
+
+    const alarmDescription = (description: string) =>
+      `Impact - ${description}. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit`;
+
+    const alarmEnabledInProd = this.withStageDependentValue({
+      app,
+      variableName: "AlarmEnabledInProd",
+      stageValues: {
+        [Stage.CODE]: false,
+        [Stage.PROD]: true,
+      },
+    });
+
+    new GuAlarm(this, "NoHealthyInstancesAlarm", {
+      app,
+      alarmName: alarmName("no healthy instances for support-frontend"),
+      alarmDescription: alarmDescription(
+        "Cannot sell any subscriptions or contributions products"
+      ),
+      actionsEnabled: alarmEnabledInProd,
+      threshold: 0.5,
+      evaluationPeriods: 2,
+      comparisonOperator: ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric: new Metric({
+        metricName: "HealthyHostCount",
+        namespace: "AWS/ApplicationELB",
+        dimensionsMap: {
+          LoadBalancer: ec2App.loadBalancer.loadBalancerFullName,
+          TargetGroup: ec2App.targetGroup.targetGroupFullName,
+        },
+        statistic: "Average",
+        period: Duration.seconds(60),
+      }),
+      snsTopicName: "reader-revenue-dev",
+    });
+
+    new GuAlarm(this, "ReducedHealthyInstancesAlarm", {
+      app,
+      alarmName: alarmName(
+        "reduced number healthy instances for support-frontend"
+      ),
+      alarmDescription: alarmDescription(
+        "Imminent issue cannot sell any subscriptions or contributions products"
+      ),
+      actionsEnabled: alarmEnabledInProd,
+      threshold: 2,
+      evaluationPeriods: 2,
+      comparisonOperator: ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric: new Metric({
+        metricName: "HealthyHostCount",
+        namespace: "AWS/ApplicationELB",
+        dimensionsMap: {
+          LoadBalancer: ec2App.loadBalancer.loadBalancerFullName,
+          TargetGroup: ec2App.targetGroup.targetGroupFullName,
+        },
+        statistic: "Average",
+        period: Duration.seconds(300),
+      }),
+      snsTopicName: "reader-revenue-dev",
+    });
+
+    new GuAlarm(this, "High5XXRateAlarm", {
```

Comment:
FYI - the pattern provides a helper for creating a high 5XX alarm. There is an example of minimal configuration for that [here](https://github.com/guardian/cdk/blob/d1de578b5691d83afa591a5b6c47cb91349ea70d/src/patterns/ec2-app/base.ts#L65-L75). The custom descriptions look important here, but I think you should be able to pass that in via the relevant alarm's [props](https://guardian.github.io/cdk/interfaces/constructs_cloudwatch.Http5xxAlarmProps.html).

At the moment, the pattern uses a % based alarm and combines two of your alarms (`High5XXRateAlarm` and `HighELB5XXRateAlarm`) into a single alarm. If using absolute thresholds and retaining the granularity of two alarms is important for your team, this might not be helpful for you.

The advantages of using the alarm provided via the pattern would be:

* reduction of boilerplate in this stack definition
* you will pick up improvements 'for free' if/when we establish best practices for monitoring/alerting in this area
* you can standardise this alarm across all of your apps more easily

However, as this is not a like for like replacement for you, it might be something to consider in a future PR.

Reply:
I think this would be a great improvement for all of those reasons that you stated. As it's not like for like I'll add a health card to address as a follow up so as to not block this PR going out